src/stripe-payment.js

```diff
@@ -0,0 +1,220 @@
+      label: { type: String },
+      country: { type: String },
+      currency: { type: String },
+      paymentMethod: { type: Object },
+      error: { type: Object },
+      output: { type: Object },
+      unsupported: { type: Boolean },
+      submitDisabled: { type: Boolean },
+      paymentText: { type: String, attribute: "payment-text" },
+      debug: { type: Boolean }
+    };
+  }
+  /**
+   * HTMLElement
+   */
+  constructor() {
+    super();
+    this.submitDisabled = true;
+    this.debug = false;
+    this.displayItems = [];
+    this.shippingOptions = [];
+    this.amount = 0;
+    this.paymentText = "Submit";
+    this.label = "Purchase";
+    this.country = "US";
+    this.currency = "usd";
+    import("@power-elements/stripe-elements/stripe-payment-request.js");
+  }
+  /**
+   * LitElement render
+   */
+  render() {
+    return html`
+      <stripe-payment-request
+        ?hidden="${this.output || this.unsupported}"
+        publishable-key="${ifDefined(
+          this.publishableKey ? this.publishableKey : undefined
+        )}"
+        .shippingOptions="${this.shippingOptions}"
+        .displayItems="${this.displayItems}"
+        amount="${this.amount}"
+        label="${this.label}"
+        country="${this.country}"
+        currency="${this.currency}"
+        @payment-method="${this.onPaymentMethod}"
+        @error="${this.onError}"
+        @success="${this.onSuccess}"
+        @fail="${this.onFail}"
+        @ready="${this.onReady}"
+        @unsupported="${this.onUnsupported}"
+        generate="source"
+        request-payer-name
+        request-payer-email
+        request-payer-phone
+      >
+        ${this.displayItems.map(
+          item => html`
+            <stripe-display-item
+              data-amount="${item.amount}"
+              data-label="${item.label}"
+            ></stripe-display-item>
+          `
+        )}
+        ${this.shippingOptions.map(
+          ship => html`
+            <stripe-shipping-option
+              data-id="${ship.id}"
+              data-label="${ship.label}"
+              data-detail="${ship.detail}"
+              data-amount="${ship.amount}"
+            ></stripe-shipping-option>
+          `
+        )}
+      </stripe-payment-request>
+      ${this.debug
+        ? html`
+            <json-viewer
+              .object="${ifDefined(
+                this.error ? this.error : this.paymentMethod
+              )}"
+            ></json-viewer>
+          `
+        : ``}
+      ${this.unsupported
+        ? html`
+            <stripe-elements
+              ?hidden="${this.output || !this.unsupported}"
+              generate="source"
+              publishable-key="${ifDefined(
+                this.unsupported ? this.publishableKey : undefined
+              )}"
+              @change="${this.onChange}"
+              @source="${this.onSuccess}"
+              @error="${this.onError}"
+            >
+            </stripe-elements>
+            <mwc-button
```

Comment:
similarly here, I think i'd rather expose `.submit` on `<stripe-payment>` and have it call element's submit, rather than injecting opinions about ui